src/app_manager/app.py

```diff
@@ -126,7 +127,18 @@ def find_resource(resource):
     p, a = roslib.names.package_resource_name(resource)
     if not p:
         raise ValueError("Resource is missing package name: %s"%(resource))
-    matches = roslib.packages.find_resource(p, a)
+
+    # roslib.packages.find_resource is too slow
+    # matches = roslib.packages.find_resource(p, a)
+    if rospack is None:
+        rospack = rospkg.RosPack()
+    pkg_path = rospack.get_path(p)
+    matches = []
+    for dirpath, dirnames, filenames in os.walk(pkg_path):
+        for filename in filenames:
+            if filename == a:
+                matches.append(os.path.join(dirpath, filename))
```

Comment:
Are you sure this works well for devel as well as install spaces, isolated and merged?
`find_resources` has a rather elaborate implementation for several layouts. That's likely for historical reasons as well as they mention "the new catkin build system", but probably not exclusively?

Also I just found `find_resources` supports an optional `rospack` parameter and explicitly caches in it! What happens if you don't change anything but make a global rospack instance to reuse here?

Reply:
I didn't understand the `find_resources`.
I will change that.

Reply:
I change to use `find_resource` with `rospack`, and it takes 4 seconds.

```
[ERROR] [1640619177.785652]: total time: 4.10533118248
```